hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMDatanodeProtocolServer.java

```diff
@@ -241,6 +241,9 @@ public SCMRegisteredResponseProto register(
                       pipelineReportsProto));
     }
     try {
+      Preconditions.checkState(registeredCommand.getError()
+              == SCMRegisteredResponseProto.ErrorCode.success,
+          "DataNode has higher Software Layout Version than SCM.");
```

Comment:
While this one-liner is elegant from code perspective, I think we should avoid polluting the audit log with the exception.

Reply:
@wzhallright Would you mind adding a slightly different check, which does not log an exception unnecessarily?